Mage.Sets/src/mage/cards/i/IanMalcolmChaotician.java

```diff
@@ -0,0 +1,267 @@
+}
+
+class IanMalcolmChaoticianCastEffect extends AsThoughEffectImpl {
+
+    IanMalcolmChaoticianCastEffect() {
+        super(AsThoughEffectType.PLAY_FROM_NOT_OWN_HAND_ZONE, Duration.WhileOnBattlefield, Outcome.PlayForFree);
+        staticText = "During each player's turn, that player may cast a spell from among the cards they don't own " +
+                "exiled with {this}";
+    }
+
+    private IanMalcolmChaoticianCastEffect(final IanMalcolmChaoticianCastEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public IanMalcolmChaoticianCastEffect copy() {
+        return new IanMalcolmChaoticianCastEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return true;
+    }
+
+    @Override
+    public boolean applies(UUID sourceId, Ability source, UUID affectedControllerId, Game game) {
+        if (!game.isActivePlayer(affectedControllerId) || IanMalcolmChaoticianWatcher.checkUsed(source, game)) {
+            return false;
+        }
+        Card card = game.getCard(CardUtil.getMainCardId(game, sourceId));
+        if (card == null || card.isLand(game)){
+            return false;
+        }
+        MageObjectReference sourceMOR = new MageObjectReference(source.getSourceId(), game);
+        return !card.getOwnerId().equals(affectedControllerId)
+                && IanMalcolmChaoticianWatcher.checkExile(sourceMOR, card, game, 0);
+    }
+}
+
+class IanMalcolmChaoticianManaEffect extends AsThoughEffectImpl implements AsThoughManaEffect {
+
+    IanMalcolmChaoticianManaEffect() {
+        super(AsThoughEffectType.SPEND_OTHER_MANA, Duration.WhileOnBattlefield, Outcome.Benefit);
+        staticText = ", and mana of any type can be spent to cast it";
+    }
+
+    private IanMalcolmChaoticianManaEffect(final IanMalcolmChaoticianManaEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return true;
+    }
+
+    @Override
+    public IanMalcolmChaoticianManaEffect copy() {
+        return new IanMalcolmChaoticianManaEffect(this);
+    }
+
+    @Override
+    public boolean applies(UUID sourceId, Ability source, UUID affectedControllerId, Game game) {
+        if (!game.isActivePlayer(affectedControllerId) || IanMalcolmChaoticianWatcher.checkUsed(source, game)) {
+            return false;
+        }
+        MageObjectReference sourceMOR = new MageObjectReference(source.getSourceId(), game);
+        Card card = game.getCard(CardUtil.getMainCardId(game, sourceId));
+        if (card == null) {
+            return false;
+        }
+        if (game.getState().getZone(card.getId()) == Zone.EXILED) {
+            return IanMalcolmChaoticianWatcher.checkExile(sourceMOR, card, game, 0);
+        }
+        CardState cardState;
+        if (card instanceof ModalDoubleFacedCard) {
+            cardState = game.getLastKnownInformationCard(((ModalDoubleFacedCard) card).getLeftHalfCard().getId(), Zone.EXILED);
```

Comment:
also I'm concerned about this instanceof callout, maybe split cards would also be affected? perhaps `getMainCard()` would work more generally?